client/landing/stepper/index.tsx

```diff
@@ -137,16 +124,40 @@ window.AppBoot = async () => {
 	setStore( reduxStore, getStateFromCache( userId ) );
 	onDisablePersistence( persistOnChange( reduxStore, userId ) );
 	setupLocale( user, reduxStore );
+	const { receiveCurrentUser } = dispatch( USER_STORE ) as UserActions;
 
-	user && initializeCalypsoUserStore( reduxStore, user as CurrentUser );
+	if ( user ) {
+		initializeCalypsoUserStore( reduxStore, user as CurrentUser );
+		receiveCurrentUser( user as UserStore.CurrentUser );
+	}
 
 	initializeAnalytics( user, getSuperProps( reduxStore ) );
 
 	setupErrorLogger( reduxStore );
 
 	const flowLoader = determineFlow();
-	const { default: rawFlow } = await flowLoader();
-	const flow = rawFlow.__experimentalUseBuiltinAuth ? enhanceFlowWithAuth( rawFlow ) : rawFlow;
+	let { default: flow } = await flowLoader();
+	let flowSteps = 'initialize' in flow ? await flow.initialize() : null;
+
+	/**
+	 * When `initialize` returns false, it means the app should be killed (the user probably issued a redirect).
+	 */
+	if ( flowSteps === false ) {
+		return;
+	}
+
+	// Checking for initialize implies this a V2 flow.
+	// CLEAN UP: once the `onboarding` flow is migrated to V2, this can be cleaned up to only support V2
+	// The `onboarding` flow is the only flow that uses in-stepper auth so far, so all the auth logic catering V1 can be deleted.
+	if ( 'initialize' in flow && flowSteps ) {
```

Comment:
Do we need to check for `initialize` here? If we have flowSteps then we already checked in
`let flowSteps = 'initialize' in flow ? await flow.initialize() : null;`
right?

Reply:
Yes but TS is not that smart. This is for types.